docs/usage/other_features/zeppelin_context.md

```diff
@@ -0,0 +1,230 @@
+
+### Object Exchange
+`ZeppelinContext` extends map and it's shared between Scala and Python environment.
+So you can put some objects from Scala and read it from Python, vice versa.
+
+<div class="codetabs">
+  <div data-lang="scala" markdown="1">
+
+{% highlight scala %}
+// Put object from scala
+%spark
+val myObject = ...
+z.put("objName", myObject)
+
+// Exchanging data frames
+myScalaDataFrame = ...
+z.put("myScalaDataFrame", myScalaDataFrame)
+
+val myPythonDataFrame = z.get("myPythonDataFrame").asInstanceOf[DataFrame]
+{% endhighlight %}
+
+  </div>
+  <div data-lang="python" markdown="1">
+
+{% highlight python %}
+# Get object from python
+%spark.pyspark
+myObject = z.get("objName")
+
+# Exchanging data frames
+myPythonDataFrame = ...
+z.put("myPythonDataFrame", postsDf._jdf)
+
+myScalaDataFrame = DataFrame(z.get("myScalaDataFrame"), sqlContext)
+{% endhighlight %}
+
+  </div>
+</div>
+
+### Form Creation
+
+`ZeppelinContext` provides functions for creating forms.
+In Scala and Python environments, you can create forms programmatically.
+<div class="codetabs">
+  <div data-lang="scala" markdown="1">
+
+{% highlight scala %}
+%spark
+/* Create text input form */
+z.input("formName")
+
+/* Create text input form with default value */
+z.input("formName", "defaultValue")
+
+/* Create select form */
+z.select("formName", Seq(("option1", "option1DisplayName"),
+                         ("option2", "option2DisplayName")))
+
+/* Create select form with default value*/
+z.select("formName", "option1", Seq(("option1", "option1DisplayName"),
+                                    ("option2", "option2DisplayName")))
+{% endhighlight %}
+
+  </div>
+  <div data-lang="python" markdown="1">
+
+{% highlight python %}
+%spark.pyspark
+# Create text input form
+z.input("formName")
+
+# Create text input form with default value
+z.input("formName", "defaultValue")
+
+# Create select form
+z.select("formName", [("option1", "option1DisplayName"),
+                      ("option2", "option2DisplayName")])
+
+# Create select form with default value
+z.select("formName", [("option1", "option1DisplayName"),
+                      ("option2", "option2DisplayName")], "option1")
+{% endhighlight %}
+
+  </div>
+</div>
+
+In sql environment, you can create form in simple template.
+
+```sql
+%spark.sql
+select * from ${table=defaultTableName} where text like '%${search}%'
+```
+
+To learn more about dynamic form, checkout [Dynamic Form](../usage/dynamic_form/intro.html).
+
+### Interpreter-Specific Functions
+
+Some interpreters use a subclass of `BaseZepplinContext` augmented with interpreter-specific
+functions. For example the dependency loader (%spark.dep) has `addRepo()`, `load()`, and other
+related methods that implement its functionality. Interpreter-specific functions are
+described within each interpreter's documentation.
+
+## Usage in Non-Programming Cells
+
+Cells with a non-programming interpreter (see table below) use a pattern match-and-replace 
```

Comment:
I think the term "non-programming interpreter" is quite confusing.

Reply:
especially sql, jdbc, pig, python, are on the list.
or even cassandra, elasticsearch

Reply:
Used a different approach. Is this better?